src/plugins/moreTags.ts

```diff
@@ -0,0 +1,240 @@
+                    label: "Only in memeber list and profiles",
+                    value: "not-chat"
+                }, {
+                    label: "Never",
+                    value: "never"
+                }]
+            }
+        ]))
+    },
+    patches: [
+        // add tags to the tag list
+        {
+            find: '.BOT=0]="BOT"',
+            replacement: [
+                {
+                    match: /(\i)\[.\.BOT=0\]="BOT";/,
+                    replace: (orig, types) =>
+                        // e[e.BOT=0]="BOT";
+                        `${tags.map((t, i) =>
+                            `${types}[${types}.${t.name}=${i * 10 + 100}]="${t.name}";\
+${t.botAndOpCases ? `${types}[${types}.${t.name}_OP=${i * 10 + 101}]="${t.name}_OP";\
+${types}[${types}.${t.name}_BOT=${i * 10 + 102}]="${t.name}_BOT";` : ""}`
+                        ).join("")}${orig}`
+                },
+                {
+                    match: /case (\i)\.BOT:default:(\i)=(.{1,20})\.BOT/,
+                    replace: (orig, types, text, strings) =>
+                        `${tags.map(t =>
+                            // case r.SERVER: T = c.Z.Messages.BOT_TAG_SERVER; break;
+                            `case ${types}.${t.name}:${text}="${t.displayName}";break;\
+${t.botAndOpCases ? `case ${types}.${t.name}_OP:${text}=${strings}.BOT_TAG_FORUM_ORIGINAL_POSTER+" • ${t.displayName}";break;\
+case ${types}.${t.name}_BOT:${text}=${strings}.BOT_TAG_BOT+" • ${t.displayName}";break;` : ""}`
+                        ).join("")}${orig}`
+                },
+                // show OP tags correctly
+                {
+                    match: /(\i)=(\i)===\i\.ORIGINAL_POSTER/,
+                    replace: "$1=$self.isOPTag($2)"
+                }
+            ],
+        },
+        // in messages
+        {
+            find: ".Types.ORIGINAL_POSTER",
+            replacement: {
+                match: /return null==(\i)\?null:\(0,/,
+                replace: "$1=$self.getTag({...arguments[0],origType:$1,location:'chat'});$&"
+            }
+        },
+        // in the member list
+        {
+            find: ".renderBot=function(){",
+            replacement: {
+                match: /this.props.user;return null!=(\i)&&.{0,10}\?(.{0,50})\(\)\.botTag/,
+                replace: "this.props.user;var type=$self.getTag({...this.props,origType:$1.bot?0:null,location:'not-chat'});\
+return type!==null?$2().botTag,type"
+            }
+        },
+        // in profiles
+        {
+            find: ",botType:",
+            replacement: {
+                match: /,botType:(\i\((\i)\)),/,
+                replace: ",botType:$self.getTag({user:$2,channelId:arguments[0].moreTags_channelId,origType:$1,location:'not-chat'}),"
+            }
+        },
+        ...passChannelIdDownProps
+    ],
+
+    getPermissions(user: User, channel: Channel): string[] {
+        if (!channel) return [];
+        const guild = GuildStore.getGuild(channel.guild_id);
```

Comment:
```suggestion
        const guild = GuildStore.getGuild(channel?.guild_id);
```
if you do this, you can delete the above channel check